GhidraPython/goresym_rename.py

```diff
@@ -0,0 +1,174 @@
+        if pclntab is not None:
+            # With pclntab, offset should be a matter of TabMeta VA
+            tmva = hints["TabMeta"]["VA"]
+            text = currentProgram.memory.getBlock(".text")
+            return text.start.offset-tmva
+
+    return None
+
+def _func_map_estimator(hints):
+    offs = []
+    # Try to guess with known function names
+    grs_funcs = hints["UserFunctions"] + hints["StdFunctions"]
+    # Match function lists and collect possible offsets
+    for f in fm.getFunctions(True):
+        name = f.name
+        if "FUN_" in name or name == "entry":
+            continue
+        for gf in grs_funcs:
+            if name in gf["FullName"]:
+                offs.append(f.entryPoint.offset-gf["Start"])
+
+    winner, winner_count = None, 0
+    # Count given offsets, most counts is our estimate
+    for o in offs:
+        count = offs.count(o)
+        print("Found offset {} with count {}".format(o, count))
+        if winner is None or winner_count < count:
+            winner, winner_count = o, count
+
+    return winner
+
+def estimate_offset(choice, hints):
+    if choice == CHOICE_EST_KNOWN_FUNCS:
+        estimator = _func_map_estimator
+    elif choice == CHOICE_EST_PCLNTAB:
+        estimator = _pclntab_estimator
+    else:
+        estimator = _entry_addr_estimator
+
+    e = estimator(hints)
+    if e is not None:
+        return e
+
+    return 0    
+
+def ask_offset(default):
+    msg = "Please specify optional memory offset for entrypoints, script estimated {} as most probable".format(default)
+    offset = askString("Optional offset", msg, str(default))
+    return int(offset)
+
+def rename_funcs(items, offset):
+    if not iterable(items):
+        return
+
+    for func in items:
+        try:
+            # ' ' is considered as invalid char
+            # https://github.com/NationalSecurityAgency/ghidra/blob/c19276091f274a9ef0850c904c743f61c850854e/Ghidra/Framework/SoftwareModeling/src/main/java/ghidra/program/model/symbol/SymbolUtilities.java#L104
+            addr, name = func["Start"], func["FullName"].replace(" ", "_")
+            entrypoint = toAddr(addr+offset)
+            f = fm.getFunctionAt(entrypoint)
+            if f is None:
+                print("Creating new func at {}".format(entrypoint))
+                createFunction(entrypoint, name)
+            else:
+                print("Renaming func {} to {}".format(f.name, name))
+                f.setName(name, USER_DEFINED)
+        except Exception as e:
+            print(u"error renaming {} at {}: {}".format(name, hex(addr), e))
+
+def annotate(items, offset):
+    if not iterable(items):
+        return
+
+    for i in items:
+        if i["VA"] != 0:
+            # ' ' is considered as invalid char
+            createLabel(toAddr(i["VA"]+offset), i["Str"].replace(" ", "_"), True)
+
+
+grsfile = askFile("GoReSym output file", "Choose GoReSym output file")
+with open(grsfile.absolutePath, "r") as fp:
+    buf = fp.read()
```

Comment:
recommend using binary mode (`rb`) and decoding explicitly or using text mode with an `encoding=utf-8` kwarg